ocw/lib/gke.py

```diff
@@ -0,0 +1,77 @@
+import os
+import subprocess
+from datetime import datetime, timezone
+import kubernetes
+import googleapiclient.discovery
+from google.oauth2 import service_account
+from ocw.lib.gce import GCE
+
+
+class GKE(GCE):
+    __instances = {}
+
+    def __new__(cls, vault_namespace):
+        if vault_namespace not in GKE.__instances:
+            GKE.__instances[vault_namespace] = self = object.__new__(cls)
+            self.__gke_client = None
+            self.__kubectl_client = {}
+            self.init()
+        return GKE.__instances[vault_namespace]
+
+    def get_project_id(self):
+        return self.get_data()["project_id"]
+
+    def gke_client(self):
+        if self.__gke_client is None:
+            self.private_key_data = self.get_data()
+            credentials = service_account.Credentials.from_service_account_info(self.private_key_data)
+            self.__gke_client = googleapiclient.discovery.build('container', 'v1', credentials=credentials)
+
+        return self.__gke_client
+
+    def kubectl_client(self, zone, cluster):
+        cluster_name = cluster["name"]
+        kube_dir = "/root/.kube"
```

Comment:
this will imply running it under root , maybe you would get username under which PCW is running and calculate home folder from that ? 

In some very far future I want to drop this approach running PCW under root , here you introducing extra  problem which we will need to solve when we will do this

Reply:
solved